spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/MutableAggregateChange.java

```diff
@@ -75,31 +61,33 @@
 	}
 
 	/**
-	 * Factory method to create an {@link MutableAggregateChange} for deleting entities.
+	 * Factory method to create a {@link MergedAggregateChange} for saving entities.
 	 *
 	 * @param entityClass aggregate root type.
 	 * @param <T> entity type.
-	 * @return the {@link MutableAggregateChange} for deleting the root {@code entity}.
-	 * @since 1.2
+	 * @return the {@link MergedAggregateChange} for saving root entities.
+	 * @since 3.0
 	 */
-	static <T> MutableAggregateChange<T> forDelete(Class<T> entityClass) {
-		return forDelete(entityClass, null);
+	static <T> MergedAggregateChange<T, AggregateChangeWithRoot<T>> mergedSave(Class<T> entityClass) {
```

Comment:
Let's keep batching/`MergedAggregateChange` outside of this type for a clear separation of batching vs. non-batching.